### PR TITLE
cite.curie module for compact identifiers via identifiers.org

### PR DESCRIPTION
**Pull request by dhimmel**
refs https://github.com/manubot/manubot/issues/218
refs https://github.com/manubot/manubot/issues/220

### CODE REVIEW

**Comment by dhimmel** [comment]
Currently stuck regarding how to deal with prefix capitalization as per https://github.com/identifiers-org/identifiers-org.github.io/issues/100

File: manubot/cite/curie/__init__.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+import requests
+
+_keep_namespace_fields = {
+    "prefix",
+    "mirId",  # MIRIAM Registry identifier
+    "name",
+    "pattern",  # regex pattern
+    "description",
+    "sampleId",  # example identifier
```
Comment (cthoyt) [inline manubot/cite/curie/__init__.py:32]: Another key you should keep is `namespaceEmbeddedInLui`. This signifies namespaces whose pattern starts with the prefix. For example, the GO pattern is `^GO:\d{7}$`, but to get a proper resolution you should only write it once like https://identifiers.org/GO:0006915 and *not like* https://identifiers.org/GO:GO:0006915

File: manubot/cite/curie/__init__.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        )
+    # do not yet understand capitlaization
+    # https://github.com/identifiers-org/identifiers-org.github.io/issues/100
+    prefix_lower = prefix.lower()
+    if prefix_lower not in get_prefixes():
+        raise ValueError(
+            f"prefix {prefix_lower} for {curie} is not a recognized prefix"
+        )
+    resolver_url = "https://identifiers.org"
+    return f"{resolver_url}/curie"
```
Comment (cthoyt) [inline manubot/cite/curie/__init__.py:101]: missed squigglys?

File: manubot/cite/curie/__init__.py (hunk 1 above)
Comment (dhimmel) [reply manubot/cite/curie/__init__.py:37]: Thanks. Excluding `namespaceEmbeddedInLui` was a crucial oversight. Unfortunately, even with knowledge of `namespaceEmbeddedInLui`, I'm worried it's insufficient to resolve all CURIEs with case-insensitive prefixes as per my comments in:

- https://github.com/identifiers-org/identifiers-org.github.io/issues/100#issuecomment-614679142
- https://github.com/identifiers-org/identifiers-org.github.io/issues/99#issuecomment-614690283

**Comment by dhimmel** [comment]
@cthoyt you [mentioned](https://github.com/identifiers-org/identifiers-org.github.io/issues/99#issuecomment-614333476) you've done some related work at https://github.com/pyobo/pyobo. Thanks a lot for the feedback! Would definitely love to chat. Will reach out when I think your expertise is most valuable

**Comment by dhimmel** [comment]
I'm reducing the scope of this PR. The new scope is just to download all identifiers.org namespaces and to expose a `manubot.cite.curie.curie_to_url` function. Future PRs will support citing CURIEs. This PR doesn't affect any existing manubot functionality.

Will address any additional review comments subsequently... merging now.

**Comment by cthoyt** [comment]
@dhimmel i still owe you some updates to the ooh-na-na interface to include identifiers.org urls when applicable. sorry it's been slow getting to that

**Comment by dhimmel** [comment]
> i still owe you some updates to the ooh-na-na interface to include identifiers.org urls when applicable.

no worries. Still a good amount of development until we're ready to use it!

**Comment by agitter** [comment]
> Will address any additional review comments subsequently

I don't have any comments.  Looks like a good base for future CURIE citations and workarounds for the upstream issues.

**Pull request merged**